renderdoc/driver/gl/gl_overlay.cpp

```diff
@@ -1239,11 +1262,66 @@ ResourceId GLReplay::RenderOverlay(ResourceId texid, FloatVector clearCol, Debug
                                  stencilCopy, sub.mip);
     }
 
-    // bind the 'real' fbo to the read framebuffer, so we can blit from it
-    drv.glBindFramebuffer(eGL_READ_FRAMEBUFFER, rs.DrawFBO.name);
+    if(useBlitFramebuffer)
+    {
+      // get latest depth/stencil from read FBO (existing FBO) into draw FBO (overlay FBO)
+      SafeBlitFramebuffer(0, 0, outWidth, outHeight, 0, 0, outWidth, outHeight,
+                          GL_DEPTH_BUFFER_BIT | GL_STENCIL_BUFFER_BIT, eGL_NEAREST);
+    }
+    else
+    {
+      // Fullscreen pass with shader to read from old depth buffer and write to new depth buffer
+      drv.glDisable(eGL_BLEND);
+      drv.glDisable(eGL_SCISSOR_TEST);
+      drv.glDisable(eGL_STENCIL_TEST);
+      drv.glColorMask(GL_FALSE, GL_FALSE, GL_FALSE, GL_FALSE);
+      drv.glEnable(eGL_DEPTH_TEST);
+      drv.glDepthFunc(eGL_ALWAYS);
+      drv.glDepthMask(GL_TRUE);
+      drv.glDisable(eGL_CULL_FACE);
+      if(!IsGLES)
+        drv.glPolygonMode(eGL_FRONT_AND_BACK, eGL_FILL);
+
+      if(DebugData.overlayTexSamples > 1)
+        drv.glUseProgram(DebugData.fullScreenCopyDepthMS);
+      else
+        drv.glUseProgram(DebugData.fullScreenCopyDepth);
+      drv.glBindProgramPipeline(0);
 
-    float green[] = {0.0f, 1.0f, 0.0f, 1.0f};
-    drv.glProgramUniform4fv(DebugData.overlayProg, overlayFixedColLocation, 1, green);
+      drv.glActiveTexture(eGL_TEXTURE0);
+      drv.glBindTexture(copyBindingEnum, curDepth);
+
+      GLuint emptyVAO = 0;
+      drv.glGenVertexArrays(1, &emptyVAO);
+      drv.glBindVertexArray(emptyVAO);
+      drv.glDrawArrays(eGL_TRIANGLE_STRIP, 0, 4);
+      drv.glDeleteVertexArrays(1, &emptyVAO);
+
+      // Restore the state
```

Comment:
GL state management is pretty ugly no matter how you slice it, so this is probably fine, but this many states changed is enough that I would consider it reasonable to declare a whole new `GLRenderState` at the top of this branch to snapshot the state to restore.

It's up to you if you'd rather leave it like this or do that route, I'm fine either way.

Reply:
Updated to use `GLRenderState` to make the state restoring simpler (and more robust).
Re-tested on Windows/Linux and Android.